### PR TITLE
T6 complete

### PR DESCRIPTION
**Pull request by mlee246**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -143,6 +144,77 @@ describe('GET/api/articles', () => {
...
+        .then(() => {
+            return request(app)
+            .get('/api/articles/9/comments')
+            .expect(200)
+            .then(({body}) => {
+                const {comments} = body
+                expect(comments.length).toBe(2)
+            })
+        })
+    })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:165]: Nice, I like that you've checked multiple articles make this test even sturdier

File: __tests__/app.test.js [hunk 2]
```diff
@@ -143,6 +144,77 @@ describe('GET/api/articles', () => {
...
+        .expect(200)
+        .then(({body}) => {
+            const {comments} = body
+            comments.forEach((comment) => {
+                expect(typeof comment.comment_id).toEqual('number')
+                expect(typeof comment.votes).toEqual('number')
+                expect(typeof comment.created_at).toEqual('string')
+                expect(typeof comment.author).toEqual('string')
+                expect(typeof comment.body).toEqual('string')
+                expect(typeof comment.article_id).toEqual('number')
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:178]: You can afford to be a bit more specific about the article id in this case they will all have the article id of 1

File: __tests__/app.test.js [hunk 3]
```diff
@@ -143,6 +144,77 @@ describe('GET/api/articles', () => {
...
+            })
+        })  
+    })
+    test('200: should respond with an array of comment objects, in order of most recent comments first', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+            const {comments} = body
+            expect(comments).toBeSortedBy('created_at')
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:188]: Nice use of jest-sorted

File: __tests__/app.test.js [hunk 4]
```diff
@@ -143,6 +144,77 @@ describe('GET/api/articles', () => {
...
+        })
+    })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:190]: We need one more 200 test here, to check that you get a 200 and an empty array of comments when an article exists but has no comments 🔍

File: __tests__/app.test.js [hunk 5]
```diff
@@ -143,6 +144,77 @@ describe('GET/api/articles', () => {
...
+    test('404: should respond with an error message when article_id is of valid type, but not found', () => {
+        return request(app)
+        .get('/api/articles/9999/comments')
+        .expect(404)
+        .then(({body}) => {
+         expect(body.msg).toEqual('No article found for article_id: 9999')
+        })
+    })
+    test('404: should respond with an error message when article_id is valid, but there are no comments associated with it', () => {
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:207]: This wouldn't be an error, just an article with no comments 
You should expect a 200 and an empty array of comments 🔍

File: __tests__/app.test.js [hunk 6]
```diff
@@ -143,6 +144,77 @@ describe('GET/api/articles', () => {
...
+        })
+    })
+    test('404: should respond with an error message when article_id is of valid type, but not found', () => {
+        return request(app)
+        .get('/api/articles/9999/comments')
+        .expect(404)
+        .then(({body}) => {
+         expect(body.msg).toEqual('No article found for article_id: 9999')
+        })
+    })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:206]: Great 400 and 404 checks ✅

File: models/comments.models.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+})
+.then(() => {
+return db.query(`
+SELECT * 
+FROM comments
+WHERE comments.article_id = $1
+ORDER BY comments.created_at ASC
+;`, [article_id])
+.then(({rows:comments}) => {
+    if(comments.length < 1){
```
Comment (nc-katherine-hurst) [inline models/comments.models.js:24]: We don't want this to Promise.reject to happen here, as long as the article exists, it's allowed to have no comments

File: models/comments.models.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+FROM articles
+WHERE article_id = $1
+;`, [article_id])
+.then(({rows}) => {
+    const article = rows[0]
+    if(!article){
+        return Promise.reject({
+            status:404, msg:`No article found for article_id: ${article_id}`
+        })}
+})
```
Comment (nc-katherine-hurst) [inline models/comments.models.js:15]: Great that you're checking to see if the article exists here, do we have a function that already exists that gets an article by its id that we could use to save repeating ourselves though?

**Review by nc-katherine-hurst — COMMENTED** [review]
Hi Meabh, great work please find your feedback below 🙂
